.github/workflows/rstudio-connect.yaml

```diff
@@ -0,0 +1,89 @@
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+
+name: RStudio Connect Integration Tests
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    env:
+      RSC_VERSION: 1.8.6.2
+      RSC_LICENSE: ${{ secrets.RSC_LICENSE }}
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+
+      # TODO: should use a real action to make sure that cleanup happens on error...
+      - name: run RStudio Connect
+        run: |
+          docker run -d --rm --name=pins-connect -p 3939:3939 --privileged -e RSC_LICENSE -e RSTUDIO_CONNECT_HASTE=enabled rstudio/rstudio-connect:${RSC_VERSION}
+
+      - uses: r-lib/actions/setup-r@v1
+        id: install-r
+
+      - name: Install pak and query dependencies
+        run: |
+          install.packages("pak", repos = "https://r-lib.github.io/p/pak/dev/")
+          saveRDS(pak::pkg_deps("local::.", dependencies = TRUE), ".github/r-depends.rds")
+        shell: Rscript {0}
+
+      - name: Restore R package cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.R_LIBS_USER }}
+          key: ubuntu-18.04-${{ steps.install-r.outputs.installed-r-version }}-1-${{ hashFiles('.github/r-depends.rds') }}
+          restore-keys: ubuntu-18.04-${{ steps.install-r.outputs.installed-r-version }}-1-
+
+      - name: Install system dependencies
+        if: runner.os == 'Linux'
+        run: |
+          pak::local_system_requirements(execute = TRUE)
+          pak::pkg_system_requirements("covr", execute = TRUE)
+        shell: Rscript {0}
+
+      - name: Install dependencies
+        run: |
+          pak::local_install_dev_deps(upgrade = TRUE)
+          pak::pkg_install("rstudio/connectapi")
+        shell: Rscript {0}
+
+      - name: output docker logs
+        run: |
+          docker logs pins-connect
+
+      - name: wait for connect to start
+        run: |
+          curl -s --retry 10 --retry-connrefused http://localhost:3939
+
+      - name: initialize Connect user "admin"
+        shell: Rscript {0}
+        run: |
+          key <- connectapi:::create_first_admin(url = "http://localhost:3939", user = "admin", password = "admin0", email = "admin@example.com")
+          connectapi:::update_renviron_creds(host = key$host, api_key = key$api_key, prefix = "CONNECT")
+
+      - name: Install package
+        run: R CMD INSTALL .
+
+      # TODO: add real integration tests here
+      - name: run tests
+        shell: Rscript {0}
+        run: |
+          # this should read the env vars by default... but it is not
+          # I think that is a bug
+          # pins::board_register_rsconnect()
+
+          # reading the env vars manually
+          pins::board_register_rsconnect(server = Sys.getenv("CONNECT_SERVER"), key = Sys.getenv("CONNECT_API_KEY"))
```

Comment:
What sets these env vars?

Reply:
A few lines up (two steps up) - `connectapi:::update_renviron_creds(host = key$host, api_key = key$api_key, prefix = "CONNECT")`

It writes the `.Renviron` file so the env vars can be picked up / used.

Reply:
Long term: I am planning to create a proper action for this set up here: https://github.com/rstudio/actions